src/vos/vos_obj.c

```diff
@@ -125,6 +221,17 @@ key_punch(struct vos_object *obj, daos_epoch_t epoch, uint32_t pm_ver,
 				break;
 			}
 		}
+
+		if (rc == 0) {
+			rc = vos_propagate_check(obj, toh, ts_set, &epr,
+						 VOS_ITER_AKEY);
+			if (rc == 1) {
+				propagated = 1;
```

Comment:
I think this gets lost. As soon as you goto punch_dkey it gets set back to 0. To be honest, I'm not a huge fan of gotos so I might not be understanding exactly what's happening.

Reply:
reworked it a bit